Tests/Api/OptimizationTests.cs

```diff
@@ -0,0 +1,125 @@
+    {
+        private int testProjectId = 1; // "EnterProjectHere";
+        private string testOptimizationId = "EnterOptimizationHere";
+        private string testOrganizationId = "EnterOrganizationHere";
+        
+        [Test]
+        public void EstimateOptimization()
+        {
+            var compile = ApiClient.CreateCompile(testProjectId);
+
+            var estimate = ApiClient.EstimateOptimization(
+                projectId: testProjectId,
+                name: "My Testable Optimization",
+                target: "TotalPerformance.PortfolioStatistics.SharpeRatio",
+                targetTo: "max",
+                targetValue: null,
+                strategy: "QuantConnect.Optimizer.Strategies.GridSearchOptimizationStrategy",
+                compileId: compile.CompileId,
+                parameters: new HashSet<OptimizationParameter>
+                {
+                    new OptimizationStepParameter("atrRatioTP", 4, 5, 1, 1) // Replace params with valid optimization parameter data for test project
+                },
+                constraints: new List<Constraint>
+                {
+                    new Constraint("TotalPerformance.PortfolioStatistics.SharpeRatio", ComparisonOperatorTypes.GreaterOrEqual, 1)
+                }
+            );
+
+            Assert.IsNotNull(estimate);
+        }
+
+        [Test]
+        public void CreateOptimization()
+        {
+            var compile = ApiClient.CreateCompile(testProjectId);
+
+            var optimization = ApiClient.CreateOptimization(
+                projectId: testProjectId,
+                name: "My Testable Optimization",
+                target: "TotalPerformance.PortfolioStatistics.SharpeRatio",
+                targetTo: "max",
+                targetValue: null,
+                strategy: "QuantConnect.Optimizer.Strategies.GridSearchOptimizationStrategy",
+                compileId: compile.CompileId,
+                parameters: new HashSet<OptimizationParameter>
+                {
+                    new OptimizationStepParameter("atrRatioTP", 4, 5, 1, 1) // Replace params with valid optimization parameter data for test project
+                },
+                constraints: new List<Constraint>
+                {
+                    new Constraint("TotalPerformance.PortfolioStatistics.SharpeRatio", ComparisonOperatorTypes.GreaterOrEqual, 1)
+                },
+                estimatedCost: 0.06m,
+                nodeType: "O2-8",
+                parallelNodes: 12
+            );
+
+            Assert.IsNotNull(optimization);
+        }
+
+        [Test]
+        public void ListOptimizations()
+        {
+            var optimizations = ApiClient.ListOptimizations(testProjectId);
+            Assert.IsNotNull(optimizations);
+        }
+
+        [Test]
+        public void ReadOptimization()
+        {
+            var optimization = ApiClient.ReadOptimization(testOptimizationId);
+            Assert.IsNotNull(optimization);
```

Comment:
Cool! similar comment about the asserts, think that for list/read/estimate/create we could assert on the returned object itself too. not just null check

Reply:
The role of the asserts here are just to verify that we get a valid result back from the API to LEAN. The API itself should have its own test suite to ensure that it returns the right data to LEAN. If you're talking about the Optimization deserialization logic specifically, then I have no problem writing asserts for this.

Reply:
Asserts have been added to cover basic integration tests.